src/api/app/assets/javascripts/webui2/project_monitor.js

```diff
@@ -6,18 +6,74 @@ function setAllLinks(event) {
   $(this).closest('.dropdown-menu').find('input').prop('checked', event.data.checked);
 }
 
-function setupProjectMonitor() { // jshint ignore:line
+function statusCell(meta, statusHash, tableInfo, projectName, packageName) {
+  var info = tableInfo[meta.col - 1];
+  var repository = info[0];
+  var architecture = info[1];
+  var status = statusHash[repository][architecture][packageName] || {};
+  var code = status.code;
+  if (code === undefined) return null;
+
+  var klass = 'build-state-' + code;
+  var output = '<a ';
+  if (['succeeded', 'failed', 'building'].includes(code)) {
+    var url = '/package/live_build_log/' + projectName + '/' + packageName + '/' + repository + '/' + architecture;
+    output += 'href="' + url + '"';
+  } else {
+    var id = meta.row + '-' + meta.col;
+    output += 'href="javascript:void(0);" id="' + id + '"';
+
+    if (status.details !== undefined) {
+      if (code === 'scheduled') klass = 'text-warning';
+      output += ' data-content="' + status.details + '" data-placement="right" data-toggle="popover"';
+    }
+  }
+  output += ' class="' + klass + '">' + code + '</a>';
+  return output;
+}
+
+function initializeMonitorDataTable() {
+  var data = $('tbody').data();
+  var packageNames = data.packagenames;
+  var statusHash = data.statushash;
+  var tableInfo = data.tableinfo;
+  var projectName = data.project;
+
   initializeDataTable('#project-monitor-table', { // jshint ignore:line
     scrollX: true,
     fixedColumns: true,
     pageLength: 50,
     lengthMenu: [[10, 25, 50, 100, -1], [10, 25, 50, 100, "All"]],
+    data: packageNames,
     search: {
       regex: true,
       smart: false,
     },
-    columnDefs: [{ width: 150, targets: 0 }]
+    columnDefs: [
+      {
+        width: 150,
+        targets: 0,
+        className: 'text-left',
+        data: null,
+        render: function (packageName) {
+          var url = '/package/show/' + projectName + '/' + packageName;
+          return '<a href="' + url + '">' + packageName + '</a>';
+        }
+      },
+      {
+        targets: '_all',
+        data: null,
+        className: 'text-center',
+        render: function (packageName, type, row, meta) {
+          return statusCell(meta, statusHash, tableInfo, projectName, packageName);
```

Comment:
nitpick: You probably could hand over the whole data set given that we seem to fetch all of them [above](https://github.com/openSUSE/open-build-service/pull/6981/files#diff-52aa359322a395d7dce7d088fe03e479R40). And if there is additional data we craft the data attribute differently.

Reply:
I am not sure what you mean. Can you please elaborate?